### PR TITLE
Add switch to not store full text of scripts run

### PR DESCRIPTION
**Pull request by peckham**
Provides a solution for #354 by adding a new configuration property

### CODE REVIEW

File: product/roundhouse.core/migrators/DefaultDatabaseMigrator.cs [hunk 1]
```diff
@@ -255,12 +255,23 @@ public IEnumerable<string> get_statements_to_run(string sql_to_run)
 
         public void record_script_in_scripts_run_table(string script_name, string sql_to_run, bool run_this_script_once, long version_id)
         {
+            var hash = hash_generator.create_hash(sql_to_run, true);
+            if (!configuration.DoNotStoreScriptsRunText)
```
Comment (erikbra) [inline product/roundhouse.core/migrators/DefaultDatabaseMigrator.cs]: Isn't this one inverted? Shouldn't the `sql_to_run` be set to `null` if `DoNotStoreScriptsRunText` is set to `true`?

File: product/roundhouse.core/migrators/DefaultDatabaseMigrator.cs [hunk 2]
```diff
@@ -255,12 +255,23 @@ public IEnumerable<string> get_statements_to_run(string sql_to_run)
...
+            }
+
             Log.bound_to(this).log_a_debug_event_containing("Recording {0} script ran on {1} - {2}.", script_name, database.server_name, database.database_name);
-            database.insert_script_run(script_name, sql_to_run, hash_generator.create_hash(sql_to_run, true), run_this_script_once, version_id);
+            database.insert_script_run(script_name, sql_to_run, hash, run_this_script_once, version_id);
         }
 
         public void record_script_in_scripts_run_errors_table(string script_name, string sql_to_run, string sql_erroneous_part, string error_message, string repository_version, string repository_path)
         {
+            if (!configuration.DoNotStoreScriptsRunText)
```
Comment (erikbra) [inline product/roundhouse.core/migrators/DefaultDatabaseMigrator.cs]: Same comment. I think this one  should not be negated?

File: product/roundhouse.console/Program.cs [hunk 3]
```diff
@@ -359,6 +359,10 @@ private static void parse_arguments_and_set_up_configuration(ConfigurationProper
                 .Add("isuptodate",
                      "This option prints whether there are any database updates or not, without actually running them. Other output except errors is disabled, to make it easy to use in scripts.",
                      option => { })
+                //do not store full text of run scripts
+                .Add("donotstorescriptsruntext",
+                     "DoNotStoreScriptsRunText - This instructs RH to not store the full script text in the database. Defaults to false.",
+                     option => configuration.DoNotStoreScriptsRunText = option != null)
```
Comment (erikbra) [inline product/roundhouse.console/Program.cs]: Is it better to name it "Positive"? i.e. `StoreScriptsRunText`, and default it to `true`? I think double negations are a bit hard to relate to (see also my comments further down, I think there is a negation too much in the code).

See e.g. https://softwareengineering.stackexchange.com/a/196837 for a discussion on affirmative (isXXX) vs negative (IsNotXXX) names on boolean variables.

Feel free to disagree, by all means :)

Comment (peckham) [reply product/roundhouse.console/Program.cs]: I do agree on the affirmative naming. That's what I had initially (hence the forgotten negations), but it didn't quite feel consistent with `DoNotCreateDatabase` and `DoNotAlterDatabase`. That's why I changed it to the negative name.

I'll try to update my PR tomorrow :)

Comment (erikbra) [reply product/roundhouse.console/Program.cs]: Fair point. Maybe we should keep it as "DoNot". There are arguments for making the "unordinary" run cases a "true" flag too, if you see what I mean. So I may be convinced in both directions, really. So, keep the negative form if you like. But, please have a look at the concerns with the double negations of the flag in the code. I think the logic is reversed (but I might be wrong)

Comment (peckham) [reply product/roundhouse.console/Program.cs]: You're right about the double negations. I've fixed them in the latest commit

Comment (erikbra) [reply product/roundhouse.console/Program.cs]: Thanks. Tested it locally, seems to work great. Thanks a lot for your contribution!!

**Pull request merged**